src/replicated_secret_sharing.rs

```diff
@@ -0,0 +1,191 @@
+use std::{
+    fmt::Debug,
+    ops::{Add, Mul, Neg, Sub},
+};
+
+use crate::field::Field;
+use crate::prss::Participant;
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct ReplicatedSecretSharing<T>(T, T);
+
+impl<T: Field> ReplicatedSecretSharing<T> {
+    #[must_use]
+    pub fn construct(a: T, b: T) -> Self {
+        Self(a, b)
+    }
+}
+
+impl<T: Field> Add for ReplicatedSecretSharing<T> {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        Self(self.0 + rhs.0, self.1 + rhs.1)
+    }
+}
+
+impl<T: Field> Neg for ReplicatedSecretSharing<T> {
+    type Output = Self;
+
+    fn neg(self) -> Self::Output {
+        Self(-self.0, -self.1)
+    }
+}
+
+impl<T: Field> Sub for ReplicatedSecretSharing<T> {
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        Self(self.0 - rhs.0, self.1 - rhs.1)
+    }
+}
+
+impl<T: Field> Mul<T> for ReplicatedSecretSharing<T> {
+    type Output = Self;
+
+    fn mul(self, rhs: T) -> Self {
+        Self(rhs * self.0, rhs * self.1)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::replicated_secret_sharing::ReplicatedSecretSharing;
+
+    use crate::field::Fp31;
+
+    fn secret_share(
```

Comment:
the way I would suggest to structure this is to have a function `secret_share(secret: u8, mut rng: &Rng) -> [ReplicatedSecretSharing<Fp31>; 3]` that does the sharing `(secret -> (u8, u8, u8)` and then produces secret shares - it could eliminate the human error constructing the shares and make tests a bit easier to understand